raiden/tests/unit/transfer/mediated_transfer/test_targetstate.py

```diff
@@ -228,6 +242,54 @@ def test_handle_secretreveal():
     assert reveal.recipient == state.route.node_address
 
 
+def test_handle_onchain_secretreveal():
+    """ The target node must update the lock state when the secret is
+    registered in the blockchain.
+    """
+    amount = 3
+    block_number = 1
+    expiration = block_number + factories.UNIT_REVEAL_TIMEOUT
+    initiator = factories.HOP1
+    our_address = factories.ADDR
+    secret = factories.UNIT_SECRET
+    pseudo_random_generator = random.Random()
+
+    channel_state, state = make_target_state(
+        our_address,
+        amount,
+        block_number,
+        initiator,
+        expiration,
+    )
+    assert factories.UNIT_SECRETHASH in channel_state.partner_state.secrethashes_to_lockedlocks
+
+    offchain_secret_reveal_iteration = target.state_transition(
+        state,
+        ReceiveSecretReveal(secret, initiator),
+        channel_state,
+        pseudo_random_generator,
+        block_number,
+    )
+    assert factories.UNIT_SECRETHASH in channel_state.partner_state.secrethashes_to_unlockedlocks
```

Comment:
Maybe useless double testing, but wouldn't it make sense to also assert the secrethash is no longer in `secrethashes_to_lockedlocks` at this point?

Reply:
Added it in a commit